encode.go

```diff
@@ -139,6 +144,61 @@ func encodeSQLString(w io.Writer, in string) error {
 	return err
 }
 
+// encodeSQLComment writes a string as a query comment.
+// If a comment is already properly formatted with C-style comments (/* */), then it will be left
+// alone, otherwise the string will be formatted into a C-style comment and anything which terminates
+// the comment prematurely will be escaped.
+func encodeSQLComment(w io.Writer, in string) error {
+	// Check if the string is already a C-style comment.
+	if strings.HasPrefix(in, string(commentStart)) && strings.HasSuffix(in, string(commentEnd)) && len(in) > 3 {
+		if _, err := w.Write(commentStart); err != nil {
+			return err
+		}
+		if err := escapeCommentContents(w, in[2:len(in)-2]); err != nil {
+			return err
+		}
+		_, err := w.Write(commentEnd)
+		return err
+	}
+
+	// Otherwise, form a C-style comment with the input string.
+	if _, err := w.Write(append(commentStart, space...)); err != nil {
+		return err
+	}
+	if err := escapeCommentContents(w, in); err != nil {
+		return err
+	}
+	_, err := w.Write(append(space, commentEnd...))
+	return err
+}
+
+// escapeCommentContents writes the contents of a C-style comment. It escapes any */ strings,
+// which would prematurely terminate the comment contents and lead to a syntax error.
+func escapeCommentContents(w io.Writer, in string) error {
```

Comment:
Is there a way to do this with just a `strings.ReplaceAll(commentEnd, escapedCommentEnd)`?

Reply:
Yes, I can do that. I wanted to copy the style of escaping that was done for encoding strings, but perhaps this is a separate enough case to not match style

Reply:
(Also, I imagine this style was used in encodeString for time efficiency, but since there's only one string to replace I think that `ReplaceAll` is probably equally as efficient as manually replacing)